nautobot/core/tests/test_navigations.py

```diff
@@ -3,20 +3,45 @@
 from django.test import TestCase
 
 from nautobot.core.apps import NAV_CONTEXT_NAMES, NavContext, NavGrouping, NavItem, register_new_ui_menu_items
+from nautobot.extras.registry import registry
+
+
+class NavMenuTestCase(TestCase):
```

Comment:
I've had a long day and I might just be tired but it seems like our navbar permissions are overly complicated

Reply:
Most definitely. :-)

Reply:
But basically it boils down to a three-level hierarchy (plus buttons which are their own thing) - a tab has permissions based on all the groups it contains, and a group has permissions based on the items it contains.